File: backend/GymRecommender/models/recommendation.py
from django.db import models
from .gym import Gym
from .request import Request
from .currency import Currency
from ..enums.rec_type import RecType

class Recommendation(models.Model):
    id = models.UUIDField(primary_key=True)
    tcost = models.DecimalField(max_digits=4, decimal_places=2)
    time = models.TimeField()
    time_score = models.DecimalField(max_digits=4, decimal_places=2)
    tcost_score = models.DecimalField(max_digits=4, decimal_places=2)
    congestion_score = models.DecimalField(max_digits=4, decimal_places=2, blank=True, null=True)
    rating_score = models.DecimalField(max_digits=4, decimal_places=2, blank=True, null=True)
    total_score = models.DecimalField(max_digits=4, decimal_places=2)
    type = models.TextField(choices=RecType)
    gym = models.ForeignKey(Gym, models.DO_NOTHING)
    request = models.ForeignKey(Request, models.DO_NOTHING)
    currency = models.ForeignKey(Currency, models.DO_NOTHING)

    class Meta:
        managed = False
        db_table = 'recommendation'
        unique_together = (('gym', 'request'),)

File: backend/Models/Recommendation.cs
﻿using System.Text.Json.Serialization;
using backend.Enums;

namespace backend.Models;

public partial class Recommendation
{
    public Guid Id { get; set; }

    public decimal Tcost { get; set; }

    public TimeOnly Time { get; set; }

    public decimal TimeScore { get; set; }

    public decimal TcostScore { get; set; }

    public decimal? CongestionScore { get; set; }

    public decimal? RatingScore { get; set; }

    public decimal TotalScore { get; set; }

    public RecommendationType Type { get; set; }

    public Guid GymId { get; set; }

    public Guid RequestId { get; set; }

    public Guid CurrencyId { get; set; }

    [JsonIgnore] public virtual Currency Currency { get; set; } = null!;

    [JsonIgnore] public virtual Gym Gym { get; set; } = null!;

    [JsonIgnore] public virtual Request Request { get; set; } = null!;
}